File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        int n = nums.size(), i = 0;
        long long answer = 0;
        
        while(i < n)
        {
            if(nums[i] == 0)
            {
                int j = i;
                long long count = 0;
                while(j < n and nums[j] == 0)
                {
                    count++, j++;
                }
                
                i = j;
                answer += count * (count + 1) / 2;
            }
            
            i++;
        }
        
        return answer;
    }
};

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        res = 0
        l, r = 0, 0
        while l < len(nums):
            if nums[r] == 0:
                while r < len(nums) and nums[r] == 0:
                    r += 1
                res += (r - l) * (r - l + 1) // 2
            else:
                r += 1
            l = r
        
        return res